filebeat/inputsource/udp/server_test.go

```diff
@@ -0,0 +1,75 @@
+package udp
+
+import (
+	"net"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const maxMessageSize = 20
+const timeout = time.Second * 15
+
+type info struct {
+	message []byte
+	addr    net.Addr
+}
+
+func TestReceiveEventFromUDP(t *testing.T) {
+	if runtime.GOOS == "window" {
+		t.Skipf("Skipped on windows")
+	}
+
+	tests := []struct {
+		name     string
+		message  []byte
+		expected []byte
+	}{
+		{
+			name:     "Sending a message under the MaxMessageSize limit",
+			message:  []byte("Hello world"),
+			expected: []byte("Hello world"),
+		},
+		{
+			name:     "Sending a message over the MaxMessageSize limit will truncate the message",
+			message:  []byte("Hello world not so nice"),
+			expected: []byte("Hello world not so n"),
+		},
+	}
+
+	ch := make(chan info)
+	host := "0.0.0.0:10000"
```

Comment:
I recommend listening only on localhost to avoid opening the port up to the world. Secondly, since this is a test that shouldn't fail if 10000 is already in use you can use `localhost:0` and it will bind to an ephemeral port. You'll need to expose the local address from the server so you know what port that it is bound to.



> If the port in the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a port number is automatically chosen. The LocalAddr method of PacketConn can be used to discover the chosen port.

https://golang.org/pkg/net/#ListenPacket